WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/AddLocalMediaToPostUseCase.kt

```diff
@@ -59,19 +59,30 @@ class AddLocalMediaToPostUseCase @Inject constructor(
 
         if (site.hasFreePlan) {
             uriList.map {
-                val videoDurationAllowed = mediaUtilsWrapper.isVideoFile(it) &&
-                        mediaUtilsWrapper.isAllowedVideoDurationForFreeSites(context, it)
+                val isVideo = mediaUtilsWrapper.isVideoFile(it)
 
-                if (videoDurationAllowed) {
+                if (isVideo) {
+                    val videoDurationAllowed = mediaUtilsWrapper.isAllowedVideoDurationForFreeSites(context, it)
+                    if (videoDurationAllowed) {
+                        result = processMediaUri(
+                                uriList,
+                                site,
+                                freshlyTaken,
+                                editorMediaListener,
+                                doUploadAfterAdding,
+                                trackEvent
+                        )
```

Comment:
NP: Could DRY this up a bit and make the condition cleaner by something like (pseudo code):

```
if ( ! isVideo || isAllowedDuration ) {
  processMediaUri
} else {
  show warning
}
```

But, for the sake of this PR (targeting a release branch) this is not a blocker, since this fixes a high impact bug.

Reply:
Thank you.  Will keep that in mind